make/Makefile

```diff
@@ -0,0 +1,103 @@
+# Copyright 2021 The cert-manager Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+SHELL := /usr/bin/env bash
+.SHELLFLAGS := -uo pipefail -c
+
+SOURCES := $(shell find . -type f -name "*.go")
+
+GOFLAGS := -ldflags '-w -s' -trimpath
+
+# GOBUILDPROCS is passed to GOMAXPROCS when running go build; if you're running make in parallel
+# using "-jN" then you'll probably want to reduce the value of GOBUILDPROCS or else you could end
+# up running N parallel invocations of go build, each of which will spin up as many threads as are
+# available on your system.
+GOBUILDPROCS ?=
+
+# Set this as an environment variable to enable signing commands using cmrel
+# Format should be:
+# projects/<project>/locations/<location>/keyRings/<keyring>/cryptoKeys/<keyname>/cryptoKeyVersions/<keyversion>
+CMREL_KEY ?= ""
+
+HOST_OS = $(shell $(GO) env GOOS)
+HOST_ARCH = $(shell $(GO) env GOARCH)
+
+include make/git.mk
+include make/tools.mk
+include make/base_images.mk
+include make/cmctl.mk
+include make/server.mk
+include make/containers.mk
+include make/release_containers.mk
+include make/manifests.mk
+include make/licenses.mk
+
+.PHONY: all
+all: binaries helm-chart static-manifests all-containers
+
+.PHONY: all-signed
+all-signed: all
+	$(MAKE) -f make/Makefile signed-artifacts
+
+.PHONY: binaries
+binaries: server-binaries cmctl kubectl-cert_manager
+
+.PHONY: signed-artifacts
+signed-artifacts: helm-chart-signature
+
+.PHONY: staged-release
+staged-release: all-signed release-manifests release-containers
+	$(MAKE) -f make/Makefile bin/release/metadata.json
```

Comment:
Adding comments here for targets which we would expect a human to use would really help. I am also a fan of the Makefile magic used [here](https://github.com/cert-manager/approver-policy/blob/0c1601b1ee6cc533ff201bedd20c4b61e0444d94/Makefile#L28) which greatly improves UX on `make help`.